src/email/email.js

```diff
@@ -0,0 +1,120 @@
+        }
+    }
+    return getUserIdorAlias(localpartRcptTo);
+};
+const splitAt = index => x => [x.slice(0, index), x.slice(index+1)];
+
+const getUserIdorAlias = function(localPart) {
+
+    // Received room+<roomalias>_hs
+    if (localPart.startsWith('room+')) {
+        log.info("Message destination is a room");
+        let alias = localPart.substring(localPart.indexOf('+')+1);
+        if (alias.lastIndexOf('_') >= 1) {
+            let res = splitAt(alias.lastIndexOf('_'))(alias);
+            res.unshift("room");
+            return res;
+        }
+      return null;
+    }
+
+    // Received user+<username>_<hs>
+    else if (localPart.startsWith('user+')) {
+        log.info("Message destination is a user");
+        let uname = localPart.substring(localPart.indexOf('+')+1);
+        if (uname.lastIndexOf('_') >= 1) {
+            return splitAt(uname.lastIndexOf('_'))(uname).unshift("user");
+        }
+      return null;
+    }
+
+    log.warn("Destination is not valid");
+    return null;
+};
+
+exports.startSMTP = function (config) {
+    const SMTP = new SMTPServer({
+        secure: false,
+        logger: false,
+        disabledCommands: ['AUTH', 'STARTTLS'],
+        authOptional: true,
+
+        onData: function (stream, session, callback) {
+            let subject, text;
+            const mailparser = new MailParser();
+            const fromAdd = ParseEmailAddress.parseOneAddress(session.envelope.mailFrom.address);
+            let receivedAddress = roomAlias(session.envelope.rcptTo, config.bridge.domain);
+            let alias = "";
+            if (!receivedAddress) {
+                return;
+            }
+            else if (receivedAddress[0] === "room") {
+                alias = `#${receivedAddress[1]}:${receivedAddress[2]}`;
+            }
+            else if (receivedAddress[0] === "user") {
+                // TODO: send message to user
+                return;
+            }
+            mailparser.on('headers', headers => {
+                subject = headers.get('subject');
+            });
+
+            mailparser.on('data', data => {
+                if (data.type === 'text') {
+                    text = data.text;
+                }
+            });
+
+            mailparser.on('end', () => {
+                if (!text.trim().length) {
+                    // text only contains whitespace (ie. spaces, tabs or line breaks)
+                    log.warn("Inbound email contains whitespace only");
+                    return;
+                }
+                log.info("Inbound email contents: "+ text.substring(0, 10) + "... ");
+
+                // The complete event MUST NOT be larger than 65535 bytes.
+                // Using 63k as the maximum text size.
+                let message = Buffer.from(text, "utf-8");
+
+                const intent = bridge.getIntent(`@_email_${fromAdd.local }_${fromAdd.domain}:${config.bridge.domain}`);
+
+                if (message.byteLength > 63000) {
```

Comment:
I'd add a const at the top of the file called `const MAX_MATRIX_`MESSAGE_SIZE = 63000` and then reference it here (and below). Generally this means that if you want to increase the size, you only do it in one place. It also means less chance of mistakes :)